src/App.test.jsx

```diff
@@ -0,0 +1,46 @@
+import { render, screen } from '@testing-library/react';
+import { MemoryRouter } from 'react-router-dom';
+import App from './App';
+import { CharacterProvider } from './context/CharacterContext';
+import userEvent from '@testing-library/user-event';
+
+describe('tests App', () => {
+  it('checks for loading, then rendered list, then navigation to a details page', async () => {
+    render(
+      <CharacterProvider>
+        <MemoryRouter>
+          <App />
+        </MemoryRouter>
+      </CharacterProvider>
+    );
+    screen.getByText(/loading/i);
+    const name = await screen.findByText('Rick Sanchez');
+    expect(name).toBeInTheDocument();
+    userEvent.click(name);
+  });
```

Comment:
Remember when testing behavior, we will also want to ensure that the detail page is loaded subsequently after clicking the Rick Sanchez link.